transformer_engine/pytorch/module/base.py

```diff
@@ -512,6 +501,16 @@ def set_tensor_parallel_group(self, tp_group: Union[dist_group_type, None]) -> N
         self.tp_group = tp_group
         self.tp_group_initialized = True
 
+    def get_fp8_params(self) -> Union[List[torch.Tensor], None]:
+        """returns the FP8 weights."""
+        fp8_params = []
+        for param in self.parameters():
+            if isinstance(param, Float8Tensor) and param.requires_grad:
+                fp8_params.append(param)
+        if len(fp8_params) == 0:
+            return None
```

Comment:
```suggestion
```
I would find it unintuitive to call a function called `get_fp8_params` and recieve a `None`. I see that this kludge is intended to work around the API for `add_fp8_tensors_to_global_buffer`, but we can make a few changes so it can take an empty list in `fp8_params`.

https://github.com/NVIDIA/TransformerEngine/blob/2d923f5b489189ec93fdc2eca86ff96f9a652c9a/transformer_engine/pytorch/fp8.py#L209
```python
            if forward and fp8_weights:
```

https://github.com/NVIDIA/TransformerEngine/blob/2d923f5b489189ec93fdc2eca86ff96f9a652c9a/transformer_engine/pytorch/fp8.py#L223
```python
                forward, bool(fp8_weights), fp8_meta["recipe"], fp8_meta["fp8_group"])
```

Alternatively, we could change this function name to `_get_fp8_params` to indicate it's an internal function.